model/core/prelude.sail

```diff
@@ -132,6 +132,27 @@ function to_bits_truncate(l, n) = get_slice_int(l, n, 0)
 val to_bits_unsafe : forall 'l, 'l >= 0 . (implicit('l), int) -> bits('l)
 function to_bits_unsafe(l, n) = get_slice_int(l, n, 0)
 
+enum Signedness = { Signed, Unsigned }
+enum VectorHalf = { High, Low }
+
+val mult_to_bits_half : forall 'l, 'l > 0 . (implicit('l), Signedness, Signedness, bits('l), bits('l), VectorHalf) -> bits('l)
+function mult_to_bits_half(l, sign1, sign2, rs1_bits, rs2_bits, high) = {
```

Comment:
We should just rename `high` too:

```suggestion
function mult_to_bits_half(l, sign1, sign2, rs1_bits, rs2_bits, result_part) = {
```

Otherwise it's a bit weird to have `high = Low`. `signed_rs1 = Unsigned` is kind of similar but reads a bit better (and `rs1_signedness` is just quite awkward).